Automated cherry pick of #108752: Fix: Clean job tracking finalizer from orphan pods

alculquicondor opened the pull request:
Cherry pick of #108752 on release-1.23.

#108752: Fix: Clean job tracking finalizer from orphan pods
#109025: Reduce number of pods in Job+GC tests

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
Fix a 1.23 regression where the job controller might not remove the job tracking finalizer from pods when deleting a job, or when the pod is orphan
```

alculquicondor committed:
Add integration test for orphan pods when there is GC

Change-Id: I04cd70725fd1830be8daf2dca53f67bc10a379b7

alculquicondor committed:
Fix: Clean job tracking finalizer from orphan pods

Change-Id: I04cd70725fd1830be8daf2dca53f67bc10a379b7

alculquicondor committed:
Remove finalizer when orphaned

Change-Id: Id88a28755660812a274dffab2693cb8a0ef4235c

alculquicondor committed:
Add test for Background delete propagation

Change-Id: I033e6fb04933c64cfe6490d1019333745d58c423

alculquicondor commented:
/kind bug
/priority critical-urgent

soltysh reviewed (approved):
/priority backlog
/triage accepted
/lgtm
/approve
/kind bug

soltysh commented:
/lgtm

alculquicondor commented:
/hold
there are concerns that the integration test is too heavy. I'm going to reduce the load.

alculquicondor committed:
Reduce number of pods in Job+GC tests

To reduce the load of the integration tests. This change reduces the runtime of each test in half.

Change-Id: I71bcaadf3809643c63bb0f6b73c28778d37d8967

alculquicondor commented:
Also added #109020 

/hold cancel

alculquicondor commented:
/retest

soltysh reviewed (approved):
/lgtm

alculquicondor commented:
/cc kubernetes/release-managers

alculquicondor commented:
/retest

alculquicondor commented:
/retest

alculquicondor commented:
/retest

alculquicondor commented:
/priority critical-urgent

alculquicondor commented:
ping @kubernetes/release-managers

alculquicondor commented:
/retest

alculquicondor commented:
/retest
